en/companion_computer/pixhawk_rpi.md

```diff
@@ -0,0 +1,259 @@
+
+https://docs.ros.org/en/humble/Installation/Ubuntu-Install-Debians.html
+
+Next, install the git in the RPi's terminal:
+
+```
+sudo apt install git
+```
+
+Next, install the XRCE_DDS standalone:
+
+```
+git clone https://github.com/eProsima/Micro-XRCE-DDS-Agent.git
+cd Micro-XRCE-DDS-Agent
+mkdir build
+cd build
+cmake ..
+make
+sudo make install
+sudo ldconfig /usr/local/lib/
+```
+
+## XRCE_DDS communication
+
+If you want to use `TELEM1`, you have to stop mavlink and then activate the microdds_client. You can do that changing the parameters:
+
+1. To disable mavlink on `TELEM1`,
+
+```
+MAV_0_CONFIG = 0
+```
+2. To start microdds_client on `TELEM1`,
+
+```
+XRCE_DDS_0_CFG = TELEM1
+```
+
+3. Use SER_TEL1_BAUD to set the baudrate.
+
+If you want to use `TELEM2` (ttyS3 of Pixhawk):
+
+```
+XRCE_DDS_0_CFG = TELEM2
+SER_TEL2_BAUD = 921600
+```
+
+Ps.: At this moment, the Pixhawk must be connected with the laptop, via `USB`, so the QGroundControl can be used to change the parameters described above.
+
+To check the status of the microdds_client, run in the QGroundControl's MavlinkConsole: 
+
+```
+microdds_client status
+```
+
+If it is not running yet, start the client in the MavlinkConsole: 
+
+```
+microdds_client start -t serial -d /dev/ttyS3 -b 921600
+```
+
+In the command above we have used the `/dev/ttyS3` to establish the XRCE_DDS communication, since it is related to `TELEM2`, according to the table:
+
+<img src="../../assets/companion_computer/pixhawk_rpi/devs.png" width="360" height="353" /> 
+
+To start the agent, run the following command in the RPi's terminal:
+
+```
+sudo MicroXRCEAgent serial --dev /dev/serial0 -b 921600
+```
+
+To see the ROS 2 topics available, run in a new RPi's terminal:
+
+```
+source /opt/ros/humble/setup.bash
+ros2 topic list
+```
+
+To communicate simultaneously via both XRCE_DDS and MAVlink, we can keep the `TELEM2` connected via RX/TX and also connect Pixhawk `USB` at the RPi. 
+
+Before disconnect the `USB` from laptop, remember to set the parameters:
+
+```
+MAV_0_CONFIG = TELEM1
+SER_TEL1_BAUD = 57600
+XRCE_DDS_0_CFG = TELEM2
+SER_TEL2_BAUD = 921600
+```
+
+The `TELEM2` is going to keep the XRCE_DDS communication with the microdds_client, as above, and the `USB` is going to keep the MAVlink communication. Remember to open a new RPi's terminal and run:
+
+```
+sudo chmod a+rw /dev/ttyACM0
+sudo mavproxy.py --master=/dev/ttyACM0 --baudrate 57600
+```
+This is a particular setup when we are looking for both offboard control, via XRCE_DDS with the RPi, and simultaneously to keep monitoring the drone via MAVlink with QGroundControl.
+Since the Pixhawk is connected with the RPi, a remote control via QGroundControl (in a ground station laptop) can be done using SSH and the mavros package, running in a new terminal:
+
+```
+ssh -C -Y user@raspberrypi_IP
+sudo chmod a+rw /dev/ttyACM0 
+roslaunch mavros px4.launch fcu_url:=/dev/ttyACM0 gcs_url:=udp://@laptop_IP
```

Comment:
Why MAVROS? Is this because you want to use ROS1? It's fine if so, I just want to understand the setup.

Reply:
> Why MAVROS? Is this because you want to use ROS1? It's fine if so, I just want to understand the setup.

I did not find in the past another way to do this bridge in the same time from the Pixhawk to the RPi and to the Laptop. This is important because the offboard control is on the RPi, but when We are developing new codes, is very important to have multiple controls. So the QGroundStation, in my laptop, can send commands to land, for example, if something goes wrong with my offboard code. 

In this context, mavros have been the best choice to setup this bridge. Since this is a very common need for the users, I prefer to explain it in the end of the Tutorial.

Reply:
Ah, that is what is confusing - we need to very clearly separate our different config setups. 

The setup you're actually likely to want is:
- Pixhawk with RPi via TELEM2 (xrce)  - ROS2 companion control
- Pixhawk with XXXXX via MAVLink, where XXXX might be RPI or QGC or whatever. It really depends if you need your RPi to get MAVLink from QGC or cloud or whatever and it is the channel for passing through MAVLink to PX4. 

Let me see if I can pull the apart a bit more.

Reply:
Yes, I agree that it can be improved to better clarify the different setups. Let me know if are still any confusion in what I meant.